### PR TITLE
Docs: `azurerm_private_endpoint` request message length correction

### LINKED ISSUE #25590: Documentation Update - azurerm_private_endpoint
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.7.2

### AzureRM Provider Version

3.58.0

### Affected Resource(s)/Data Source(s)

azurerm_private_endpoint

### Terraform Configuration Files

```hcl
In resource:
azurerm_private_endpoint

Parameter:
request_message

Doco says: (Optional) A message passed to the owner of the remote resource when the private endpoint attempts to establish the connection to the remote resource. The request message can be a maximum of 140 characters in length. Only valid if is_manual_connection is set to true.

In particular the `max length of 140 chars in length`

This is not accurate. The actual max length is 128 chars.

variable "request_message" {
  description = "(Optional) A message passed to the owner of the remote resource when the private endpoint attempts to establish the connection to the remote resource. The request message can be a maximum of 140 characters in length. Only valid if is_manual_connection is set to true."
  type        = string
  default     = "Please approve this connection."

  validation {
    condition     = length(var.request_message) <= 140
    error_message = "Request Message must be less than or equal to 140 characters."
  }
}

My neat validation early warning system failed me today.
```

### Debug Output/Panic Output

```shell
Error: creating Private Endpoint "xyz" (Resource Group "ghj"): network.PrivateEndpointsClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="RequestMessageTooLong" Message="Call to Microsoft.Sql/servers failed. Error message: Request Message may only be a maximum of 128 characters" Details=[]
```

### Expected Behaviour

Expect the doco to say 128 chars please.

### Actual Behaviour

Doco says 140 chars.

### Steps to Reproduce

visit:

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/private_endpoint

Be lied to.,

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by ASHR4**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description

Documentation is currently providing the wrong information for the `request_message` attribute with regards to it's length

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [X] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [X] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [X] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [X] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [X] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_private_endpoint` - amend request message limit documentation

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [X] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #25590

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: website/docs/r/private_endpoint.html.markdown [hunk 1]
```diff
@@ -238,7 +238,7 @@ A `private_service_connection` block supports the following:
 
 -> **NOTE:** Some resource types (such as Storage Account) only support 1 subresource per private endpoint.
 
-* `request_message` - (Optional) A message passed to the owner of the remote resource when the private endpoint attempts to establish the connection to the remote resource. The request message can be a maximum of `140` characters in length. Only valid if `is_manual_connection` is set to `true`.
+* `request_message` - (Optional) A message passed to the owner of the remote resource when the private endpoint attempts to establish the connection to the remote resource. The request message can be a maximum of `128` characters in length. Only valid if `is_manual_connection` is set to `true`.
```
Comment (stephybun) [inline website/docs/r/private_endpoint.html.markdown:241]: In the validation for this field we allow up to `140` characters. Local testing confirms that this limit is valid and suggests that this limit varies depending on the service the private endpoint is connected to. Unfortunately I can't find any documentation on this

Could we instead rephrase this to 
```suggestion
* `request_message` - (Optional) A message passed to the owner of the remote resource when the private endpoint attempts to establish the connection to the remote resource. The provider allows a maximum request message length of `140` characters, however the request message maximum length is dependent on the service the private endpoint is connected to. Only valid if `is_manual_connection` is set to `true`.
```

Alternatively we can leave the description as it currently is and add a note below:

```suggestion
* `request_message` - (Optional) A message passed to the owner of the remote resource when the private endpoint attempts to establish the connection to the remote resource. The request message can be a maximum of `140` characters in length. Only valid if `is_manual_connection` is set to `true`.

-> **NOTE:** When connected to a SQL resource the `resource_message` maximum length is `128`.
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this PR @ASHR4. Would you addressing the comment left in-line? Once that's done this should be good to go.

File: website/docs/r/private_endpoint.html.markdown (hunk 1 above)
Comment (deitChi) [reply website/docs/r/private_endpoint.html.markdown:241]: Personally prefer a mix between if I may have an input:

```
* `request_message` - (Optional) A message passed to the owner of the remote resource when the private endpoint attempts to establish the connection to the remote resource. The provider allows a maximum request message length of `140` characters, however the request message maximum length is dependent on the service the private endpoint is connected to. Only valid if `is_manual_connection` is set to `true`.

**EXAMPLE:** When connected to a SQL resource the `resource_message` maximum length is `128`, however MS does not provide docs for service mapping to char limit.
```

Would that work for you?

Comment (stephybun) [reply website/docs/r/private_endpoint.html.markdown:241]: If we can tweak the `**EXAMPLE:**` line to the following for consistency, as well as removing the last sentence since it isn't providing actionable information for the user, then that would be fine

```-> **NOTE:** When connected to a SQL resource the `resource_message` maximum length is `128`.```

File: website/docs/r/private_endpoint.html.markdown [hunk 2]
```diff
@@ -240,6 +240,8 @@ A `private_service_connection` block supports the following:
 
 * `request_message` - (Optional) A message passed to the owner of the remote resource when the private endpoint attempts to establish the connection to the remote resource. The request message can be a maximum of `140` characters in length. Only valid if `is_manual_connection` is set to `true`.
```
Comment (stephybun) [inline website/docs/r/private_endpoint.html.markdown:241]: @deitChi had requested the following change as well
```suggestion
* `request_message` - (Optional) A message passed to the owner of the remote resource when the private endpoint attempts to establish the connection to the remote resource. The provider allows a maximum request message length of `140` characters, however the request message maximum length is dependent on the service the private endpoint is connected to. Only valid if `is_manual_connection` is set to `true`.
```

**Review by stephybun — CHANGES_REQUESTED** [review]

**Review by stephybun — APPROVED** [review]
Thanks @ASHR4 LGTM 👍
